service/controller/appcatalog/resource/appcatalogentry/create.go

```diff
@@ -137,7 +160,13 @@ func (r *Resource) newAppCatalogEntries(ctx context.Context, cr v1alpha1.AppCata
 			r.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("failed to parse latest version for %#q in catalog %#q", name, cr.Name), "stack", fmt.Sprintf("%#v", err))
 		}
 
-		for _, entry := range entries {
+		maxEntries := r.maxEntriesPerApp
+		if len(entries) < maxEntries {
```

Comment:
if entries are shorter than the maximum limitation, use the length of entries instead.